tests/unit/schema/test_schema_logic.py

```diff
@@ -0,0 +1,202 @@
+            ("integer", "integer"),
+            ("number", "number"),
+            ("text", "string"),
+            ("path", "string"),
+            ("choice", "string"),
+            ("filename", "string"),
+            ("directory", "string"),
+            ("LIST", "array"),
+            (None, "string"),
+        ]
+    )
+    def test_param_formatted_correctly(self, param_type, expected_type):
+        mock_param = MagicMock()
+        mock_param.name = "param_name"
+        mock_param.type.name = param_type
+        mock_param.help = "param description"
+        mock_param.default = None
+
+        formatted_param = format_param(mock_param)
+
+        self.assertIsInstance(formatted_param, SamCliParameterSchema)
+        self.assertEqual(formatted_param.name, "param_name")
+        self.assertEqual(formatted_param.type, expected_type)
+        self.assertEqual(formatted_param.description, "param description")
+        self.assertEqual(formatted_param.default, None)
+
+    @parameterized.expand(
+        [
+            ("list", SamCliParameterSchema("p_name", "array", default="default value", items="string")),
+            ("choice", SamCliParameterSchema("p_name", "string", default=["default", "value"], choices=["1", "2"])),
+        ]
+    )
+    @patch("schema.schema.isinstance")
+    def test_param_formatted_given_type(self, param_type, expected_param, isinstance_mock):
+        mock_param = MagicMock()
+        mock_param.name = "p_name"
+        mock_param.type.name = param_type
+        mock_param.type.choices = ["1", "2"]
+        mock_param.help = None
+        mock_param.default = ("default", "value") if param_type == "choice" else "default value"
+        isinstance_mock.return_value = True if param_type == "choice" else False  # mock check against click.Choice
+
+        formatted_param = format_param(mock_param)
+
+        self.assertEqual(expected_param, formatted_param)
+
+    @patch("schema.schema.isinstance")
+    @patch("schema.schema.format_param")
+    def test_getting_params_from_cli_object(self, format_param_mock, isinstance_mock):
+        mock_cli = MagicMock()
+        mock_cli.params = []
+        param_names = ["param1", "param2", "config_file", None]
+        for param_name in param_names:
+            mock_param = MagicMock()
+            mock_param.name = param_name
+            mock_cli.params.append(mock_param)
+        format_param_mock.side_effect = lambda x: x.name
+
+        params = get_params_from_command(mock_cli)
+
+        self.assertIn("param1", params)
+        self.assertIn("param2", params)
+        self.assertNotIn("config_file", params)
+        self.assertNotIn(None, params)
+
+    @parameterized.expand([(False,), (True,)])
+    @patch("schema.schema.importlib.import_module")
+    @patch("schema.schema.get_params_from_command")
+    @patch("schema.schema.isinstance")
+    def test_command_structure_is_retrieved(self, is_group, isinstance_mock, get_params_mock, import_mock):
```

Comment:
Should this test case be in it's own test class with two test cases? Just poking at this since we are testing "regular commands", and commands with sub commands in the same test with `if` checks.

Reply:
We could, but I thought to implement it this way since there would be a lot of overlapping logic with the setup and assertions (i.e. the initialization of the `mock_module` and iterating through `commands` to verify its contents). Either way would work, but this way there's a little less duplication

Reply:
How about we make it more modular? One option would be to utilize the setup method to initialize anything we might need for all the tests. Then for validation we could have a helper method that does that. This way, if something were to change in the future, we could pinpoint what needs to be changed more quickly.

Reply:
Yup, makes sense 👍